include/segment_symbols.h

```diff
@@ -24,9 +24,11 @@
 #define SEGMENT_END(segment) (_ ## segment ## SegmentEnd)
 #define SEGMENT_SIZE(segment) ((uintptr_t)SEGMENT_END(segment) - (uintptr_t)SEGMENT_START(segment))
 
-#define SEGMENT_ROM_START(segment) (_ ## segment ## SegmentRomStart)
-#define SEGMENT_ROM_END(segment) (_ ## segment ## SegmentRomEnd)
-#define SEGMENT_ROM_SIZE(segment) ((uintptr_t)SEGMENT_ROM_END(segment) - (uintptr_t)SEGMENT_ROM_START(segment))
+#define SEGMENT_ROM_START(segment) ((uintptr_t) _ ## segment ## SegmentRomStart)
+#define SEGMENT_ROM_START_OFFSET(segment, offset) ((uintptr_t) (( _ ## segment ## SegmentRomStart ) + (offset)))
+#define SEGMENT_ROM_END(segment) ((uintptr_t) _ ## segment ## SegmentRomEnd)
+#define SEGMENT_ROM_SIZE(segment) (SEGMENT_ROM_END(segment) - SEGMENT_ROM_START(segment))
+#define SEGMENT_ROM_SIZE2(segment) ((size_t)( _ ## segment ## SegmentRomEnd - _ ## segment ## SegmentRomStart ))
```

Comment:
I'm not really sure what to do here, and to a large extent it depends on what we decide to do with types for rom addresses and segmented addresses. I don't like the `SEGMENT_ROM_START_OFFSET` macro, though, it seems unnecessary, it looks too much like getting the offset of something in a struct, when really it's just the most basic pointer arithmetic operation.

Reply:
`SEGMENT_ROM_START_OFFSET` is necessary because of some places in the codebase requiring some arithmetic before the `uintptr_t` cast or it wouldn't match.

I added the cast as part of the macros to avoid having to use the cast explicitly on every use of `SEGMENT_ROM_START` and `SEGMENT_ROM_END`

Reply:
It's not necessary in that it wasn't required before, though, it's only become necessary because of the way you currently have the macros arranged, surely?

Reply:
It was an issue tho, there were some awful casting in place already
![image](https://github.com/zeldaret/mm/assets/7416381/9a681e5a-ac6a-49dc-9c0e-931eb64d5d9e)